core/src/main/scala/com/nvidia/spark/rapids/tool/EventLogPathProcessor.scala

```diff
@@ -101,7 +106,74 @@ object EventLogPathProcessor extends Logging {
         isDBEventLogFile(path.getName)
       }
     })
-    (dbLogFiles.size > 1)
+    dir.isDirectory && dbLogFiles.size > 1
+  }
+
+  /**
+   * Identifies if the input file or directory is a valid event log.
+   *
+   * TODO - Need to handle size of files in directory, for now document its not supported.
+   *        Reference: https://github.com/NVIDIA/spark-rapids-tools/pull/1275
+   *
+   * @param s   FileStatus to be identified.
+   * @param fs  FileSystem instance for file system operations.
+   * @return    Option[EventLogInfo] if valid, None otherwise.
+   */
+  private def identifyEventLog(s: FileStatus, fs: FileSystem): Option[EventLogInfo] = {
+    if (s.isFile && eventLogNameFilter(s.getPath)) {
+      // Regular event log file. See function `eventLogNameFilter` for criteria.
+      Some(ApacheSparkEventLog(s.getPath))
+    } else if (isEventLogDir(s)) {
+      // Apache Spark event log directory (starting with "eventlog_v2_").
+      Some(ApacheSparkEventLog(s.getPath))
+    } else if (isDatabricksEventLogDir(s, fs)) {
+      // Databricks event log directory (files starting with "eventlog").
+      Some(DatabricksEventLog(s.getPath))
+    } else {
+      // Ignore other types of files.
+      None
+    }
+  }
+
+  /**
+   * Retrieves all event logs from the input path.
+   *
+   * Note:
+   *   - If the input path is a directory and recursive search is enabled, this function will
+   *     search for event logs in all subdirectories.
+   *   - This is done using a queue to avoid stack overflow.
+   *
+   * @param inputPath               Root path to search for event logs.
+   * @param fs                      FileSystem instance for file system operations.
+   * @param recursiveSearchEnabled  If enabled, search for event logs in all subdirectories.
+   * @return                        List of (EventLogInfo, EventLogFileSystemInfo) tuples.
+   */
+  private def getEventLogInfoInternal(inputPath: Path, fs: FileSystem,
+      recursiveSearchEnabled: Boolean): List[(EventLogInfo, EventLogFileSystemInfo)] = {
+    val results = ArrayBuffer[(EventLogInfo, EventLogFileSystemInfo)]()
+    val queue = mutable.Queue[FileStatus]()
+    queue.enqueue(fs.getFileStatus(inputPath))
+    while (queue.nonEmpty) {
```

Comment:
this doesn't work when the --no-recursion option is specified and you point to a directory of event logs, that needs to be handled.

Reply:
Thanks for catching the root level case. Fixed it.